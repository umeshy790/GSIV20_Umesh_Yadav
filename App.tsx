/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React from 'react';
import UpcomingMovies from './src/components/UpcomingMovies';
import {Provider} from 'react-redux';
import {store} from './src/store';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import MovieDetails from './src/components/MovieDetails';

const Stack = createStackNavigator();

const App: React.FC = () => (
  <>
    <Provider store={store}>
      <NavigationContainer>
        <Stack.Navigator initialRouteName="Home">
          <Stack.Screen
            name="Home"
            options={{header: () => null}}
            component={UpcomingMovies}
          />
          <Stack.Screen
            name="MovieDetails"
            options={{title: 'Back'}}
            component={MovieDetails}
          />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  </>
);

export default App;
