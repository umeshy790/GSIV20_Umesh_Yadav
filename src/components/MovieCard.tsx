import React from 'react';
import {StyleSheet, Text, View, Image, TouchableWithoutFeedback} from 'react-native';
import {Movie} from '../modal/modals';

interface Prop {
  movie: Movie,
  handlePress : any
}

const MovieCard = (props: Prop) => {
  const { movie, handlePress } = props;

  const _handlePress = () => handlePress(movie)

  return (
    <TouchableWithoutFeedback style={{flex: 1}} onPress={_handlePress}>
      <View style={styles.container}>
        <View style={styles.body}>
          <Image
             style={styles.image}
             source={{
              uri: `http://image.tmdb.org/t/p/original${movie?.poster_path}`
            }}
          />
          <View style={{flexDirection : 'row', paddingTop : 5, paddingHorizontal : 8}}>
            <Text ellipsizeMode="tail" numberOfLines={1} style={styles.title}>{movie.title}</Text>
            <Text style={styles.popularity}>({movie.vote_average})</Text>
          </View>
          <View style={{paddingVertical : 5, paddingHorizontal : 8}}>
             <Text ellipsizeMode="tail" numberOfLines={2} style={styles.description}>{movie.overview}</Text>
          </View>
        </View>
      </View>
    </TouchableWithoutFeedback>
  );
};

export default MovieCard;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    borderWidth: 12,
    borderColor: "transparent"
  },
  body: {
    flex: 1,
    maxWidth: 180,
    height: 290,
    borderRadius: 20,
    backgroundColor: "#dfdfdf",
    shadowColor: "#000",
    shadowOffset: {
    	width: 0,
     height: 3,
    },
    shadowOpacity: 0.23,
    shadowRadius: 3.42,
    elevation: 6,
   },
  image: {
    borderTopLeftRadius: 20,
    borderTopRightRadius : 20,
      height: 200,
      width : undefined
  },
  title: {
    fontSize: 16,
    color: '#4285f4',
    fontWeight: '500',
    flex : 1,
  },
  popularity: {
    fontSize: 15,
    color: "#4a4a4a",
    fontWeight : "500"
  },
  description: {
    fontSize: 14,
    color : "#4a4a4a"
  }
});
