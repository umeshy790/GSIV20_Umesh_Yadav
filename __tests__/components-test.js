import React from 'react';
import renderer from 'react-test-renderer';
import MovieCard from '../src/components/MovieCard';
import App from '../App';

// global.fetch = jest.fn(() => new Promise(resolve => resolve()));
// jest.mock('react-native-gesture-handler', () => {});

describe('Component tests', () => {
  describe('App Test', () => {
    it('should match to snapShot', () => {
      const tree = renderer.create(<App />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Movie Card Test', () => {
    const movie = {
      id: 1212,
      original_language: 'en',
      original_title: 'dummy',
      overview: 121,
      release_date: 112,
      title: 'dummy',
      vote_average: 1212,
      adult: false,
      backdrop_path: '',
      genre_ids: [121, 22],
      popularity: 4.3,
      poster_path: 'fjfjd',
      video: false,
      vote_count: 121,
    };
    it('should match to snapShot', () => {
      const tree = renderer.create(<MovieCard movie={movie} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
