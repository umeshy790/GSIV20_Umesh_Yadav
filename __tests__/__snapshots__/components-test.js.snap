// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component tests Movie Card Test shoul match to snapShot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "transparent",
      "borderWidth": 12,
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "borderRadius": 20,
        "elevation": 6,
        "flex": 1,
        "height": 290,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.23,
        "shadowRadius": 3.42,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http://image.tmdb.org/t/p/originalfjfjd",
        }
      }
      style={
        Object {
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "height": 200,
          "width": undefined,
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "paddingHorizontal": 8,
          "paddingTop": 5,
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#4285f4",
            "flex": 1,
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        dummy
      </Text>
      <Text
        style={
          Object {
            "color": "#4a4a4a",
            "fontSize": 15,
            "fontWeight": "500",
          }
        }
      >
        (
        1212
        )
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 8,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Object {
            "color": "#4a4a4a",
            "fontSize": 14,
          }
        }
      >
        121
      </Text>
    </View>
  </View>
</View>
`;
